Lib/test/test_asyncio/test_events.py

```diff
@@ -823,6 +823,28 @@ def test_create_server(self):
         # close server
         server.close()
 
+    def test_create_server_trsock(self):
+        proto = MyProto(self.loop)
+        f = self.loop.create_server(lambda: proto, '0.0.0.0', 0)
+        server = self.loop.run_until_complete(f)
+        self.assertEqual(len(server.sockets), 1)
+        sock = server.sockets[0]
+        host, port = sock.getsockname()
+        self.assertEqual(host, '0.0.0.0')
+        dup = sock.dup()
+        self.addCleanup(dup.close)
+        self.assertIsInstance(dup, socket.socket)
+        self.assertFalse(sock.get_inheritable())
+        with self.assertRaises(ValueError):
+            sock.settimeout(1)
```

Comment:
And why can't we set the timeout on the socket?

Reply:
Because the socket is nonblocking so you cannot set the timeout.

Reply:
Oh, I see the reason now. `TransportSocket` will only allow setting the timeout to 0.